### PR TITLE
improve error messages

### PR DESCRIPTION
**Pull request by pavelzw**
https://github.com/Quantco/slim-trees/pull/74#discussion_r1283800628

### CODE REVIEW

File: slim_trees/lgbm_booster.py [hunk 1]
```diff
@@ -198,8 +198,9 @@ def _compress_booster_handle(model_string: str) -> Tuple[str, List[dict], str]:
     return front_str, trees, back_str
 
 
-def _validate_tree_structure(tree: dict) -> bool:
```
Comment (pavelzw) [inline slim_trees/lgbm_booster.py:201]: No idea how this slipped under our radar 😅 let's hope this doesn't break any tests

File: slim_trees/sklearn_tree.py [hunk 2]
```diff
@@ -141,7 +141,10 @@ def _decompress_tree_state(state: dict):
             "values",
         },
         *({"missing_go_to_left"} if sklearn_version >= Version("1.3") else set()),
-    }
+    }:
+        raise ValueError(
+            "Invalid tree structure. Do you use an unsupported scikit-learn version?"
+        )
```
Comment (StefanSorgQC) [inline slim_trees/sklearn_tree.py:147]: A failure can also occur for supported scikit-learn versions. Namely, when unpickling a 1.1 model with version 1.3. Can we add support for that?

Comment (pavelzw) [reply slim_trees/sklearn_tree.py:147]: Would it also fail in this `state.keys() != ...` part or do you want an additional check if the scikit-learn versions don't match that a `ValueError` gets thrown?
I think scikit-learn already warns you if you want to unpickle a model with a newer version than it was pickled...

Comment (StefanSorgQC) [reply slim_trees/sklearn_tree.py:147]: Yes, `sklearn` does issue a warning, but with a custom `Unpickler`, it works.

However, one would get stuck here.

**Review by StefanSorgQC — APPROVED** [review]
Looks good, thanks.

Can we (in a separate PR) relax the condition to a warning and leave it to sklearn to raise an error if necessary?

**Comment by pavelzw** [comment]
> Can we (in a separate PR) relax the condition to a warning and leave it to sklearn to raise an error if necessary?

Since we directly access every element of the state after the assertion, a warning wouldn't gain us anything since there would then be another error directly afterwards.

**Pull request merged**